js/admin/instructorNew.js

```diff
@@ -0,0 +1,221 @@
+
+    // console.log(instructor);
+    // open modal
+    handleModalOpening('Edit', instructor);
+  }else if (btn.title === 'delete' && confirm('You are about to remove the card')) {
+    // console.log('nay')
+    // get the card
+    const card = btn.parentElement.parentElement.parentElement.parentElement;
+
+    // remove it from
+    card.remove();
+  }
+}
+
+const extractCardDetail = (card) => {
+  const instructor = {};
+
+  // extracting the details
+  instructor.name = card.querySelector('.card__title').textContent;
+  instructor.role = card.querySelector('.card__subtitle').textContent;
+  instructor.bio = card.querySelector('.card__description').textContent;
+  instructor.githubID = card.querySelector('.githubID').textContent;
+  instructor.twitterID = card.querySelector('.twitterID').textContent;
+
+  return instructor;
+}
+// handle the add click
+const handleAddInstructor = (e) => {
+  handleModalOpening('Add');
+}
+
+const handleModalOpening = (action, instructor =  {
+  name: '',
+  role: '',
+  bio: '',
+  githubID: '',
+  twitterID: ''
+}) => {
+  // getting all the required details
+  // the header and btn texts
+  const modeTextSpaces = document.querySelectorAll('.dialog-mode');
+  const modeIcon = document.getElementById('update-icon');
+
+  modeIcon.className = 'fas fa-user-plus';
+  
+  // reveal the modal by removing the hide class
+  modalDialog.classList.remove('hide');
+  
+
+  // change the text to the required action
+  modeTextSpaces.forEach((space) => {
+    space.textContent = action;
+  })
+
+  // fill the fields if required
+  if (action.toLowerCase() === "edit") {
+    fillModalFields(instructor);
+
+    // change icon
+    modeIcon.className = 'fas fa-user-edit';
+  }
+}
+
+const fillModalFields = (instructor) => {
+  // the input fields
+  const nameUI = document.getElementById('instructorName'),
+  roleUI = document.getElementById('instructorRole'),
+  bioUI = document.getElementById('instructorBio'),
+  githubUI = document.getElementById('githubID'),
+  twitterUI = document.getElementById('twitterID');
+
+  const fieldList = [githubUI, twitterUI, bioUI, roleUI, nameUI]
+
+
+  // fill in the fields 
+  nameUI.value = instructor.name;
+  roleUI.value = instructor.role;
+  bioUI.value = instructor.bio;
+  githubUI.value = instructor.githubID;
+  twitterUI.value = instructor.twitterID;
+
+  // focus all fields
+  fieldList.forEach(field => {
+    field.focus();    
+  })
+}
+
+// check the clicked item
+const handleModalRemove = (e) => {
+  const target = e.target;
+
+  // if the target is any where around the document that is not the modal or it is the cancel btn, the modal should disappear;
+  if (target.className === 'mdc-dialog__scrim') {
+    hideModal();
+  }
+
+  if (target.closest('button') == null) {
+    return;
+  }
+  else if (target.closest('button').id === 'close-btn') {
+    hideModal();
+  }
+} 
+const hideModal = () => {
+  modalDialog.classList.add("hide");
```

Comment:
similarly, here you could open the modal by referring to the appropriate item under `dialogInstances`, e.g.:
```js
dialogInstances.modal.close();
```
Assuming `data-modal-id` was set to `modal` for the modal.

Reply:
done